app/helpers/requests_helper.rb

```diff
@@ -37,17 +37,26 @@ def i18n_library_title_key
     current_request.origin
   end
 
+  # rubocop:disable Metrics/MethodLength
   def label_for_item_selector_holding(holding)
     if holding.checked_out?
-      content_tag :span, class: 'status pull-right availability unavailable' do
-        "Due #{holding.due_date}"
+      if Settings.ils.bib_model == 'SearchworksItem'
+        content_tag :span, class: 'status pull-right availability unavailable' do
+          "Due #{holding.due_date}"
+        end
+      else
+        tag.span class: 'status pull-right availability unavailable' do
+          # This content is lazily loaded, because the query to get due_date makes the initial query for items take a long time.
+          turbo_frame_tag "due_date_#{holding.id}", src: due_date_path(holding.id)
```

Comment:
Should we be concerned about triggering hundreds or thousands of separate requests on our problem records? I thought turbo eagerly loaded frames by default?

Reply:
It will, but this is only when they are checked out.  Do we have many instances with lots of items and we expect them all to be checked out at once?

Reply:
Our big offenders (from SAL3) at the moment:
https://searchworks.stanford.edu/view/9984192 has 49
https://searchworks.stanford.edu/view/487166 has 27
https://searchworks.stanford.edu/view/10034758 has 24

There's a pretty good smattering of 10-20 checked out items too.

I guess given the number of API calls the FOLIO UI itself makes, it's probably fine??

Reply:
I tested this with 9984192. It takes a long time for the due dates to come in (I see requests made immediately, but nothing updates for ~10-15s). I'm surprised how much longer this is compared to the RTAC response we make to load due dates on https://searchworks-folio-dev.stanford.edu/view/9984192 (~0.5s)

Reply:
I'm not sure what you want me to do.  I think this helps us draw the initial page faster, which was the intention of #1676, right?

Reply:
I don't think the behavior demonstrated above is acceptable. I think we could look at why it takes so long for turbo to load the due date information, look at using RTAC to asynchronously load the data (as we're doing in Searchworks and seems fast enough), or consider some option.

Reply:
Updated to do just a single request for all items. This gets all the dates in 6s rather than 2-3 seconds per row.

Reply:
I'm surprised it still takes 6s. Searchworks does it in <1s. Maybe @corylown can help with using RTAC.

Reply:
@cbeer might it be because we're proxying through the graphql server and not hitting the folio server directly?